specs/deque.spec.js

```diff
@@ -0,0 +1,154 @@
+const Deque = require('../src/deque');
+
+describe('Deque', () => {
+    describe('Deque Constructor', () => {
+        it('Create new Deque', () => {
+            const Deque = new Deque();
+
+            assert.equal(Deque.data, []);
+        });
+    });
+
+    describe('Deque push function', () => {
+        beforeEach( () => { 
+            const Deque = new Deque();
+            const arr = [1, 2, 3, 4, 5];
+            Deque.push(true, arr);
```

Comment:
as `push` function is under test better to call it inside test in execution stage.